x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/Hasher.java

```diff
@@ -610,27 +612,84 @@ private static char[] getPbkdf2Hash(SecureString data, int cost, String prefix)
         }
     }
 
+    private static int parsePbkdf2Iterations(char[] hash, String prefix) {
+        int separator = -1;
+        for (int i = prefix.length(); i < hash.length; i++) {
+            if (hash[i] == '$') {
+                separator = i;
+                break;
+            }
+        }
+        if (separator == -1) {
+            throw new IllegalArgumentException("unknown hash format");
+        }
+        try {
+            return Integer.parseInt(new String(hash, prefix.length(), separator - prefix.length()));
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("hash must include a valid iteration number", e);
+        }
+    }
+
     private static boolean verifyPbkdf2Hash(SecureString data, char[] hash, String prefix) {
```

Comment:
nit: can you add some java doc explaining that for verify we can dynamically find the key size and salt length for user provided hashes, but for hashes we create we always use the get method to create  hashes with fixed key sizes and fixed salt lengths.